heartbeat/monitors/logger/logger.go

```diff
@@ -0,0 +1,113 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package logger
+
+import (
+	"errors"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+var eventLogger *logp.Logger = nil
+
+const ActionMonitorRun = "monitor.run"
+
+type durationLoggable struct {
+	Ms int64 `json:"ms"`
+}
+
+type monitorRunInfo struct {
+	MonitorID string           `json:"id"`
+	Type      string           `json:"type"`
+	Duration  durationLoggable `json:"duration"`
+	Steps     *int             `json:"steps,omitempty"`
+}
+
+func NewMonitorRunInfo(id string, t string, dMs int64) monitorRunInfo {
+	return monitorRunInfo{
+		MonitorID: id,
+		Type:      t,
+		Duration:  durationLoggable{Ms: dMs},
+	}
+}
+
+func SetLogger(l *logp.Logger) *logp.Logger {
```

Comment:
Do we even need this? Why not create it upfront anyway and remove these Setters and Getters? 
```go
var eventLogger *logp.Logger = logp.NewLogger("heartbeat.events")
```

Reply:
If we don't have a way to set the logger externally then it means we don't have access to inserting our own observed loggers for testing. I tried to find a way around this but thought the cleanest way was to have a setter here.

Do you think there's a better way to allow for that?

Reply:
I don't have any alternate suggestion right now.